contracts/erc721/OpenSeaGasFreeListing.sol

```diff
@@ -8,6 +8,18 @@ pragma solidity >=0.8.0 <0.9.0;
 
 /// @notice Library to achieve gas-free listings on OpenSea.
 library OpenSeaGasFreeListing {
+
+    /**
+    @notice Convinience function to get the right chainId
+     */
+    function getChainId() internal view returns (uint256) {
+        uint256 id;
+        assembly {
+            id := chainid()
+        }
+        return id;
+    }
```

Comment:
(minor) You can simplify this by naming the return variable as a means of declaring the `id`:

```Solidity
function getChainId() internal view returns (uint256 id) {
  assembly {
    id := chainid()
  }
}
```

For anything longer than this I think so-called "naked returns" are risky, but here it's quite tidy.

Reply:
oh nice, this is way cleaner!